AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSAnalyticsTransmissionTarget.m

```diff
@@ -71,24 +72,36 @@ - (void)trackEvent:(NSString *)eventName {
  * @param properties dictionary of properties.
  */
 - (void)trackEvent:(NSString *)eventName withProperties:(nullable NSDictionary<NSString *, NSString *> *)properties {
-  NSMutableDictionary *mergedProperties = [NSMutableDictionary new];
+  MSEventProperties *eventProperties;
+  if (properties) {
+    eventProperties = [MSEventProperties new];
+    for (NSString *key in properties.allKeys) {
+      NSString *value = properties[key];
+      [eventProperties setString:value forKey:key];
+    }
+  }
+  [self trackEvent:eventName withTypedProperties:eventProperties];
+}
+
+- (void)trackEvent:(NSString *)eventName withTypedProperties:(nullable MSEventProperties *)properties {
+  MSEventProperties *mergedProperties = [MSEventProperties new];
 
   // Merge properties in its ancestors.
   MSAnalyticsTransmissionTarget *target = self;
   while (target != nil) {
-    [target mergeEventPropertiesWith:mergedProperties];
+    [target.propertyConfigurator mergeTypedPropertiesWith:mergedProperties];
     target = target.parentTarget;
   }
 
   // Override properties.
-  if (properties) {
-    [mergedProperties addEntriesFromDictionary:(NSDictionary * _Nonnull)properties];
-  } else if ([mergedProperties count] == 0) {
+  if (properties && properties.properties) {
+    [mergedProperties.properties addEntriesFromDictionary:(NSDictionary<NSString *, MSTypedProperty *> *)properties.properties];
+  } else if ([mergedProperties.properties count] == 0) {
```

Comment:
You can use isEmpty from MSEventProperties directly :)

Reply:
👍